lib/src/main/kotlin/net/dean/jraw/references/PublicContributionReference.kt

```diff
@@ -202,4 +203,62 @@ abstract class PublicContributionReference internal constructor(reddit: RedditCl
                 ))
         }
     }
+
+    /**
+     * Sets the [spoilerStatus] of a post using the official Reddit spoiler flair.
+     *
+     * TODO - Requires mod privileges for the subreddit that contains the post.
+     */
+    @EndpointImplementation(Endpoint.POST_SPOILER, Endpoint.POST_UNSPOILER)
+    fun setPostSpoiler(spoilerStatus: Boolean) {
+        val endpoint = if (spoilerStatus) Endpoint.POST_SPOILER else Endpoint.POST_UNSPOILER
+        reddit.request {
+            it.endpoint(endpoint)
+                .post(mapOf(
+                    "id" to fullName
+                ))
+        }
+    }
+
+    /**
+     * Set or unset the announcement [state] of a post in its subreddit.
+     * Reddit provides 2 allowable announcement "slots" for a subreddit, often referred to as sticky posts.
+     *
+     * The [num] argument is optional, an integer between 1 and 3, and only used when stickying a post.
+     * It allows specifying a particular "slot" to sticky the post into,
+     * and if there is already a post stickied in that slot it will be replaced.
+     * If there is no post in the specified slot to replace, or num is None, the bottom-most slot will be used.
+     *
+     * Setting [toProfile] will sticky the post to an authenticated user's profile page.
+     * This is only used with new profile pages, not supported for legacy profiles.
+     *
+     * TODO - Requires mod privileges for the subreddit that contains the post.
+     */
+    @EndpointImplementation(Endpoint.POST_SET_SUBREDDIT_STICKY)
+    fun stickyPost(state: Boolean, num : Int = -1, toProfile: Boolean = false) {
```

Comment:
Does this also apply to comments? If yes, this can be renamed to `sticky`, `makeSticky`, or something comparable. Otherwise it should be moved to `SubredditReference`

Reply:
It just applies to posts, I think the comment version is covered under distinguish functionality. I'll move it over.